### DISCUSSION TITLE
Response format compatible with PSR-7 on phalcon 3.4 (need for RoadRunner)

### DISCUSSION THREAD

**Discussion by serhii-m-prime** [phalcon/cphalcon#15755] [Q&A]
For run application written on phalcon 3.4.2 need response format compatible with PSR-7. [RoadRunner Implementation](https://roadrunner.dev/docs/integration-phalcon)
So can I get all headers as an array from the phalcon application with a response?
e.g. looks loke doesn't work.
`$phalconResponse = $application->handle();
$phalconResponse->getHeaders()->toArray();`

**Comment by niden** [comment]
The response format is not compatible with PSR7 at the moment. This has been requested in the past and is indeed in our todo list. The whole HTTP layer will be refactored to use middleware as well as responses according to PSR.

You could use now the `Http\Message` namespace to work with your requests and responses but you will need a bit of work to get things working with that.

The refactoring will happen after we have a stable v5 (this year) and while we work on v6 which will be PHP based.

**Comment by niden** [comment]
https://github.com/phalcon/cphalcon/issues/14390

This is the original issue and it has quite a high vote count in our NFRs (things the community needs)

**Comment by niden** [comment]
Check this out also https://github.com/kcloze/phalcon-roadrunner

**Reply by serhii-m-prime** [reply]
Already checked, it works without response headers (
